os/osx.cmake

```diff
@@ -69,12 +69,24 @@ set(
     "${XCODE_DEVELOPER_ROOT}/SDKs/MacOSX${OSX_SDK_VERSION}.sdk"
 )
 
+# If OSX_SDK_VERSION is not set, the SDK version is computed using
+# xcrun. However, it is possible for the version reported by xcrun to not
+# be present in the SDKs folder. In that case, xcrun can also give the
+# full path to a valid SDK.
+execute_process(
```

Comment:
We only need to run it if `__osx_sysroot_suggestion_1` and `__osx_sysroot_suggestion_2` not present, it make sense to move it to `else()` block:
```cmake
if(EXISTS ...)
elseif(EXISTS ...)
else()
  execute_process(...)
endif()
```